src/main/java/de/tum/in/www1/artemis/domain/LtiPlatformConfiguration.java

```diff
@@ -0,0 +1,131 @@
+package de.tum.in.www1.artemis.domain;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import javax.annotation.Nullable;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+import javax.validation.constraints.NotNull;
+
+import org.hibernate.annotations.Cache;
+import org.hibernate.annotations.CacheConcurrencyStrategy;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+/**
+ * Represents the configuration for an LTI platform.
+ * Stores details such as registration ID, client ID, and various URIs needed for LTI communication.
+ */
+@Entity
+@Table(name = "lti_platform_configuration")
+@Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)
+@JsonInclude(JsonInclude.Include.NON_EMPTY)
+public class LtiPlatformConfiguration extends DomainObject {
+
+    /** Entity name for LTI platform configuration. */
+    public static final String ENTITY_NAME = "ltiPlatformConfiguration";
+
+    @NotNull
+    @Column(name = "registration_id", nullable = false)
+    private String registrationId;
+
+    @NotNull
+    @Column(name = "client_id", nullable = false)
+    private String clientId;
+
+    @Nullable
+    @Column(name = "original_url")
+    private String originalUrl;
+
+    @Nullable
+    @Column(name = "custom_name")
+    private String customName;
+
+    @NotNull
+    @Column(name = "authorization_uri", nullable = false)
+    private String authorizationUri;
+
+    @NotNull
+    @Column(name = "jwk_set_uri", nullable = false)
+    private String jwkSetUri;
+
+    @NotNull
+    @Column(name = "token_uri", nullable = false)
+    private String tokenUri;
+
+    @OneToMany(mappedBy = "ltiPlatformConfiguration", fetch = FetchType.EAGER)
```

Comment:
Do we need `EAGER` here?

Reply:
Thank you for pointing that out. We can use lazy fetch here, I just fixed it. 
Just a small note that this property would be actually used in the follow-up PR (#7734 ) where each online course would be configured with different Lti platforms.